ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/application_traffic_recognition.schema.yml

```diff
@@ -94,67 +95,19 @@ keys:
                 dest_prefix_set_name:
                   type: str
                   description: Destination prefix set name.
-                protocols:
-                  type: list
-                  description: |-
-                    List of protocols to consider for this application.
 
-                    To use port field-sets (source, destination or both), the list
-                    must contain only one or two protocols, either `tcp` or `udp`.
-                    When using both protocols, one line is rendered for each in the configuration,
-                    hence the field-sets must have the same value for `tcp_src_port_set_name` and
-                    `udp_src_port_set_name` and for `tcp_dest_port_set_name` and `udp_dest_port_set_name`
-                    if set in order to generate valid configuration in EOS.
-                  items:
-                    type: str
-                    valid_values:
-                      - ahp
-                      - esp
-                      - icmp
-                      - igmp
-                      - ospf
-                      - pim
-                      - rsvp
-                      - tcp
-                      - udp
-                      - vrrp
-                protocol_ranges:
-                  type: list
-                  description: |-
-                    Acccept protocol value(s) or range(s).
-                    Protocol values can be between 1 and 255.
-                  items:
-                    type: str
-                    convert_types:
-                      - int
-                udp_src_port_set_name:
-                  type: str
-                  description: |-
-                    Name of field set for UDP source ports.
-
-                    When the `protocols` list contain both `tcp` and `udp`, this key value
-                    must be the same as `tcp_src_port_set_name`.
-                tcp_src_port_set_name:
-                  type: str
-                  description: |-
-                    Name of field set for TCP source ports.
-
-                    When the `protocols` list contain both `tcp` and `udp`, this key value
-                    must be the same as `udp_src_port_set_name`.
-                udp_dest_port_set_name:
-                  type: str
-                  description: |-
-                    Name of field set for UDP destination ports.
-
-                    When the `protocols` list contain both `tcp` and `udp`, this key value
-                    must be the same as `tcp_dest_port_set_name`.
-                tcp_dest_port_set_name:
+          l4_applications:
+            type: list
+            description: List of user defined L4 applications.
```

Comment:
Can you add in the description that the name should not be the same as the one for an ipv4_application ? (and same for `ipv4_applicaiton`)

I tested and it gives:

```
edge4A(config-app-l4-sdsa)#application ipv4 sdsa
! Trying to edit application with incorrect type. Did you mean 'application l4 sdsa'?
```

I looked into making this better but we don't support this dictionary sublist unicity today and it bodes the question of changing the model maybe later to some unique list with a type that would be `l4` or `ipv4`
Though then we would have some keys that would not be usable in `l4` mode which is also not great.

For now lets carry on like this with a description

Reply:
Apparently this behaviour is by design (shared application namespace), as then using the defined applications the CLI has no way to distinguish between IPv4 (and IPv6) and L4.
Updated description to reflect this